ansible/roles/nginx/tasks/main.yml

```diff
@@ -0,0 +1,77 @@
+---
+- name: Add nginx repository
+  apt_repository: repo='ppa:nginx/stable' state=present
+
+- name: Install nginx
+  sudo: yes
+  apt: name={{ item }} state=present update_cache=yes cache_valid_time=3600
+  with_items: ubuntu_pkg
+  when: nginx_install == True
+
+- name: Create nginx http user
+  sudo: yes
+  user: name={{ nginx_user }} createhome=no password=no state=present groups="{{ cchq_user }}"
+  when: nginx_install == True
+
+- name: Create nginx logfile home
+  sudo: yes
+  file: path={{ log_home }} owner={{ cchq_user }} group={{ cchq_user }} mode=0770 state=directory
+  when: nginx_install == True
+
+- name: Create static files home
+  sudo: yes
+  file: path={{ nginx_static_home }} owner={{ cchq_user }} group={{ cchq_user }} mode=0755 state=directory
+  when: nginx_install == True
+
+- name: Copy the nginx configuration file
+  sudo: yes
+  template: src=nginx.conf.j2 dest=/etc/nginx/nginx.conf
+  notify:
+   - restart nginx
+
+- name: Copy the nginx default configuration file
+  sudo: yes
+  template: src=default.conf.j2 dest=/etc/nginx/conf.d/default.conf
+
+- name: Copy the nginx default site configuration file
+  sudo: yes
+  template: src=default.j2 dest=/etc/nginx/sites-available/default
+
+- name: Create the link for site enabled specific configurations
+  sudo: yes
+  file: path=/etc/nginx/sites-enabled/default state=link src=/etc/nginx/sites-available/default
+
+- name: Create the configurations for sites independent of inventory
+  sudo: yes
+  template: src=site.j2 dest=/etc/nginx/sites-available/{{ item['server']['file_name'] }}
+  with_items: nginx_sites
+  when: nginx_sites|lower != 'none'
+  notify:
+   - reload nginx
+
+- name: Create the links to enable site configurations
+  sudo: yes
+  file: path=/etc/nginx/sites-enabled/{{ item['server']['file_name'] }} state=link src=/etc/nginx/sites-available/{{ item['server']['file_name'] }}
+  with_items: nginx_sites
+  when: nginx_sites|lower != 'none'
+
+- include_vars: '{{deploy_env}}.yml'
```

Comment:
I'm not incredibly proud of this method for including variable configurations for each environment.  I chose this because:
* It let's us use one nginx template
* It let's us control site configurations by environment (e.g. ones that depend on inventory setup)
* It also lets us deploy sites that are independent of inventory (e.g. slow.)